src/quantity_api/__init__.py

```diff
@@ -5,10 +5,25 @@
 import optype as op
 
 type Unit = Any  # TODO: unit-api
+type Dimension = Any  # TODO: dimension-api
 
 __version__: Final = "0.0.1.dev0"
 __all__ = ["__version__", "Quantity"]
 
+
+@runtime_checkable
+class MetrologyNamespace[Q, U: Unit, D: Dimension](Protocol):
+    
+    @staticmethod
+    def asquantity(obj: V, unit: obj) -> Q[V, U[D]]: ...
+
+    @staticmethod
+    def asunit(obj) -> U[D]: ...
+
+    @staticmethod
+    def asdimension(obj) -> D: ...
```

Comment:
Sounds like we're converging on the minimal

```suggestion
    def asdimension(obj: str | D, /) -> D: ...
```